### PR TITLE
Add an http -> https redirect frontend and service.

### PR DESCRIPTION
**Pull request by brendandburns**
Wire it up to the ingress.

@eparis @rmmh @fejta

### CODE REVIEW

File: mungegithub/submit-queue/deployment/nginx-redirect/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,41 @@
+events {
+  worker_connections  1024;
+}
+
+http {
+  server {
+    listen 80;
+    server_name contrib.submit-queue.k8s.io;
+    location / {
+        if ($http_user_agent ~ (GoogleHC) ) {
```
Comment (fejta) [inline mungegithub/submit-queue/deployment/nginx-redirect/nginx.conf]: What is GoogleHC?

**Comment by fejta** [comment]
Any chance this regressed: https://storage.googleapis.com/kubernetes-test-history/k8s-queue-health.svg

**Comment by fejta** [comment]
(not sure if that's still updating or why the timestamp on the bottom left is red)

File: mungegithub/submit-queue/deployment/nginx-redirect/nginx-deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        image: nginx
+        ports:
+        - containerPort: 80 
+        volumeMounts:
+        - name: conf
+          mountPath: /etc/nginx
+      volumes:
+      - name: conf
+        configMap:
+          name: nginx-https-redirect
```
Comment (eparis) [inline mungegithub/submit-queue/deployment/nginx-redirect/nginx-deploy.yaml]: what is this configMap?

File: mungegithub/submit-queue/deployment/nginx-redirect/nginx.conf (hunk 1 above)
Comment (brendandburns) [inline mungegithub/submit-queue/deployment/nginx-redirect/nginx.conf]: That's the user-agent that the Google HTTP LoadBalancer sends for it's health checks.

File: mungegithub/submit-queue/deployment/nginx-redirect/nginx-deploy.yaml (hunk 2 above)
Comment (brendandburns) [inline mungegithub/submit-queue/deployment/nginx-redirect/nginx-deploy.yaml]: It contains the contents of the `nginx.conf` file in this PR. Would you like me to include the config map as well?

I created it with:

```
kubectl create cm nginx-https-redirect --from-file=nginx.conf=nginx.conf
```

(I could add a README too, I suppose)

**Comment by brendandburns** [comment]
@fejta I don't know how that graph is created, do you have a pointer to the code?

The 'sample' time certainly corresponds to when I pushed the config.

Would you like me to rollback?  (or I'm happy to fix up that picture...)

Thanks
--brendan

**Comment by fejta** [comment]
Code is here: https://github.com/kubernetes/test-infra/blob/master/queue-health/poll/poller.py

Deployment is here: https://github.com/kubernetes/test-infra/blob/master/queue-health/queue-health-prod.yaml#L27

I suspect maybe the code either does not redirect or else rejects the cert?

**Comment by brendandburns** [comment]
I added a README.md to explain the ConfigMap

I also fixed `poller.py` to handle both `https` and redirects (should they be needed in the future).

I don't think I have permissions to redeploy the poller, so either someone needs to merge/rebuild, or I need permissions to push a new version of that container.

**Comment by fejta** [comment]
Thanks! Still /lgtm

We should probably migrate this to gcr.io/k8s-testimages. I can give you access to that if you want to push the image there.

And if you merge it I'll update the deployment.

**Comment by brendandburns** [comment]
I pushed gcr.io/k8s-testimages/queue-health-poll:v20170407 with these fixes. I updated the image in the checked-in poller config as well.

Checking the logs for the running poller, things seem to be restored.

**Comment by brendandburns** [comment]
@k8s-bot bazel test this

**Pull request merged**